marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubConfigImpl.java

```diff
@@ -927,19 +927,19 @@ public void loadConfigurationFromProperties(Properties environmentProperties) {
             // could be factored away with FlowRunner
             mlUsername = getEnvPropString(environmentProperties, "mlUsername", mlUsername);
             mlPassword = getEnvPropString(environmentProperties, "mlPassword", mlPassword);
-
+            
             isHostLoadBalancer = getEnvPropBoolean(environmentProperties, "mlIsHostLoadBalancer", false);
             String mlHost = getEnvPropString(environmentProperties, "mlHost", null);
             String lbh = getEnvPropString(environmentProperties, "mlLoadBalancerHosts", null);
-            if (isHostLoadBalancer) {
-                if (mlHost != null && lbh != null && !mlHost.equals(lbh)){
-                    throw new DataHubConfigurationException("mlLoadBalancerHosts must be the same as mlHost");
-                }
-                else {
-                    loadBalancerHost = mlHost;
-                }
+            
+            if (mlHost != null && lbh != null && !mlHost.equals(lbh)){
+                throw new DataHubConfigurationException("mlLoadBalancerHosts must be the same as mlHost");
             }
-
+            else {
+                loadBalancerHost = mlHost;
+                isHostLoadBalancer = true;
```

Comment:
What if "isHostLoadBalancer" is set to FALSE in the gradle.property file?
As "mlLoadBalancerHosts" is a deprecated property the burden of a check must fall on "mlIsHostLoadBalancer".